tests/test_build.py

```diff
@@ -0,0 +1,263 @@
+NOT_FOUND_HTTP_RESPONSE.reason = "Not Found"
+NOT_FOUND_HTTP_RESPONSE._content = b'{"errors": [{"status": 404, "message": "Not found ... "}]}'
+NOT_FOUND_HTTP_RESPONSE.encoding = "utf-8"
+NOT_FOUND_HTTP_RESPONSE.url = "http://jfrog-server..."
+
+NOT_FOUND_EXCEPTION_BODY = requests.exceptions.HTTPError(response=NOT_FOUND_HTTP_RESPONSE)
+
+
+@responses.activate
+def test_get_build_info_success(mocker):
+    responses.add(
+        responses.GET,
+        f"{URL}/api/build/build_name/abc",
+        json=BUILD_INFO.model_dump(),
+        status=200,
+    )
+
+    artifactory_build = ArtifactoryBuild(AuthModel(url=URL, auth=AUTH))
+    mocker.spy(artifactory_build, "get_build_info")
+    get_build = artifactory_build.get_build_info("build_name", "abc")
+
+    assert isinstance(get_build, BuildInfo)
+
+
+@responses.activate
+@pytest.mark.parametrize(
+    "build_num,properties_input,expected_query,raised_exc",
+    [
+        ("11", BuildProperties(), "", BuildNotFoundError),
+        ("11", BuildProperties(diff="43"), "?diff=43", BuildNotFoundError),
+        ("11", BuildProperties(diff="43", started="2024-08"), "?diff=43&started=2024-08", BuildNotFoundError),
+        ("11", BuildProperties(diff="abc"), "?diff=abc", ArtifactoryError),
+        ("def", BuildProperties(diff="22"), "?diff=22", ArtifactoryError),
+    ],
+)
+def test_get_build_info_errors(mocker, build_num, properties_input, expected_query, raised_exc):
+    responses.add(
+        responses.GET,
+        f"{URL}/api/build/build_name/{build_num}{expected_query}",
+        body=NOT_FOUND_EXCEPTION_BODY,
+        status=404,
+    )
+
+    artifactory_build = ArtifactoryBuild(AuthModel(url=URL, auth=AUTH))
+    mocker.spy(artifactory_build, "get_build_info")
+    with pytest.raises(raised_exc):
+        artifactory_build.get_build_info("build_name", build_num, properties=properties_input)
+
+
+@responses.activate
+@pytest.mark.parametrize(
+    "error_model_dump,raised_exc",
+    [
+        (BUILD_NOT_FOUND_ERROR.model_dump(), BuildNotFoundError),
+        (BUILD_GENERIC_ERROR.model_dump(), ArtifactoryError),
+    ],
+)
+def test_get_build_info_response_error(mocker, error_model_dump, raised_exc):
+    responses.add(responses.GET, f"{URL}/api/build/build_name/123", body=NOT_FOUND_EXCEPTION_BODY, status=200)
+
+    artifactory_build = ArtifactoryBuild(AuthModel(url=URL, auth=AUTH))
+    mocker.spy(artifactory_build, "get_build_info")
+    with pytest.raises(raised_exc):
+        artifactory_build.get_build_info("build_name", "123")
+
+
+@responses.activate
+def test_promote_build_success(mocker):
+    responses.add(
+        responses.GET,
+        f"{URL}/api/build/build_proj/build_number",
+        json=BUILD_INFO.model_dump(),
+        status=200,
+    )
+    _promotion_request = BUILD_PROMOTION_REQUEST.model_dump()
+    responses.add(
+        responses.POST,
+        f"{URL}/api/build/promote/build_proj/build_number",
+        json=_promotion_request,
```

Comment:
I think there is a problem here with the usage of the [responses](https://github.com/getsentry/responses) library.

The `json` param is used to mock the API response, not the request. You should have instead `json=promotion_response`.